spyder/plugins/ipythonconsole.py

```diff
@@ -911,6 +911,22 @@ def update_working_directory(self):
         if shellwidget is not None:
             shellwidget.get_cwd()
 
+    def wait_shell_is_ready(self, shell, timeout=3000):
+        """Wait until console It's ready and banner is correctly displayed.
+
+        shell (ShellWidget)
+        timeout (int) milliseconds
+        """
+        i = 0
+        loop = QEventLoop()
+        timer = QTimer()
+        timer.timeout.connect(loop.quit)
+        timer.start(300)

```

Comment:
Are we waiting only 300 ms for the prompt to be ready?

Reply:
Ok, I saw above that `timeout=3000`, but this should be higher. I think we should set it to `15000` at least.

Reply:
We are waiting a maximum of 3seconds, and checking every 300ms, testing in my machine It takes ~1.5 seconds to be ready